install_dev_repos.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+#
+
+"""
+Helper script for installing spyder and external-deps locally in editable mode.
+"""
+
+import argparse
+import os
+import sys
+from logging import Formatter, StreamHandler, getLogger
+from pathlib import Path
+from subprocess import check_output
+
+from importlib_metadata import PackageNotFoundError, distribution
```

Comment:
This won't work in Python 3.7 because `importlib_metadata` was added in Python 3.8. And we still support Python 3.7.

Reply:
This is the backport (and forward-port) package `importlib_metadata`, not the stdlib `import.metadata`. Its already required and installed as a transitive dep by many of our existing deps, so its guaranteed to be present in the user's env, though I'll add it as an explicit direct dep to the necessary places in my followup PR migrating off the deprecated `pkg_resources` for entrypoints, as discussed earlier here.